RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,638 @@
+  edm::ESHandle<PixelClusterParameterEstimator> pe;
+  const PixelClusterParameterEstimator* pp;
+  iSetup.get<TkPixelCPERecord>().get(pixelCPE_, pe);
+  pp = pe.product();
+  //--------------------------end ---------------------//
+
+  edm::ESHandle<TrackerTopology> tTopoHandle;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  const TrackerTopology* const tTopo = tTopoHandle.product();
+
+  auto output = std::make_unique<edmNew::DetSetVector<SiPixelCluster>>();
+
+  int jet_number = 0;
+  for (unsigned int ji = 0; ji < cores->size(); ji++) {  //loop jet
+    jet_number++;
+
+    if ((*cores)[ji].pt() > ptMin_) {
+      std::set<long long int> ids;
+      const reco::Candidate& jet = (*cores)[ji];
+      const reco::Vertex& jetVertex = (*vertices)[0];
+
+      std::vector<GlobalVector> splitClustDirSet = splittedClusterDirections(jet, tTopo, pp, jetVertex, 1);
+      bool l2off = (splitClustDirSet.empty());
+      if (splitClustDirSet.empty()) {  //if layer 1 is broken find direcitons on layer 2
+        splitClustDirSet = splittedClusterDirections(jet, tTopo, pp, jetVertex, 2);
+      }
+      splitClustDirSet.push_back(GlobalVector(jet.px(), jet.py(), jet.pz()));
+      for (int cc = 0; cc < (int)splitClustDirSet.size(); cc++) {
+        //-------------------TensorFlow setup - tensor (2/2)----------------------//
+        tensorflow::NamedTensorList input_tensors;
+        input_tensors.resize(3);
+        input_tensors[0] =
+            tensorflow::NamedTensor(inputTensorName_[0], tensorflow::Tensor(tensorflow::DT_FLOAT, input_size_eta));
+        input_tensors[1] =
+            tensorflow::NamedTensor(inputTensorName_[1], tensorflow::Tensor(tensorflow::DT_FLOAT, input_size_pt));
+        input_tensors[2] = tensorflow::NamedTensor(inputTensorName_[2],
+                                                   tensorflow::Tensor(tensorflow::DT_FLOAT, {input_size_cluster}));
+
+        //put all the input tensor to 0
+        input_tensors[0].second.matrix<float>()(0, 0) = 0.0;
+        input_tensors[1].second.matrix<float>()(0, 0) = 0.0;
+        for (int x = 0; x < jetDimX; x++) {
+          for (int y = 0; y < jetDimY; y++) {
+            for (int l = 0; l < 4; l++) {
+              input_tensors[2].second.tensor<float, 4>()(0, x, y, l) = 0.0;
+            }
+          }
+        }
+        //-----------------end of TF setup (2/2)----------------------//
+
+        GlobalVector bigClustDir = splitClustDirSet.at(cc);
+
+        LocalPoint jetInter(0, 0, 0);
+
+        jet_eta = jet.eta();
+        jet_pt = jet.pt();
+        input_tensors[0].second.matrix<float>()(0, 0) = jet.eta();
+        input_tensors[1].second.matrix<float>()(0, 0) = jet.pt();
+
+        edmNew::DetSetVector<SiPixelCluster>::const_iterator detIt = inputPixelClusters->begin();
+
+        const GeomDet* globDet =
+            DetectorSelector(2, jet, bigClustDir, jetVertex, tTopo);  //select detector mostly hitten by the jet
+
+        if (globDet == nullptr)
```

Comment:
nothing special actually: if the bigClustDir do not cross any pixel detector modules nullptr is returned, simply no windows are opened in this direction because no clusters are released in this direction. Happens quite often, so better to not print an error message for that. I can add a comment about that at this line